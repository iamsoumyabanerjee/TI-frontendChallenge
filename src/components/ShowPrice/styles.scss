@import '../../styles/variables.scss';

.price-container {
  color: $matterhorn;
  display: flex;
  flex-direction: row;
  &.price-sale {
    .actual-price {
      text-decoration: line-through;
      margin-right: 10px;  
    }
    .sale-price {
      color: $lightRed;
      font-weight: bold;
    }
  }
  
}