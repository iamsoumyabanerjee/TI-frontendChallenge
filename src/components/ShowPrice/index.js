import React from 'react'

import styles from './styles.scss'

const ItemPrice = ({ price, finalPrice }) => {
  return (
    <div className={`price-container ${price !== finalPrice ? 'price-sale' : ''}`}>
      <span className='actual-price'>${price}</span>
      { price !== finalPrice && <span className='sale-price'>${finalPrice}</span> }
    </div>
  )
} 

export default ItemPrice