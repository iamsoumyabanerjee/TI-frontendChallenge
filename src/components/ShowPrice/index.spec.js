import React from 'react'
import renderer from 'react-test-renderer'
import ShowPrice from '.'

describe('ShowPrice', () => {
  it('should render correctly when actual price and final price are same', () => {
    const props = {
      price: 100,
      finalPrice: 100
    }
    const tree = renderer
      .create(<ShowPrice { ...props } />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
  it('should render correctly when actual price and final price are not same', () => {
    const props = {
      price: 100,
      finalPrice: 80
    }
    const tree = renderer
      .create(<ShowPrice { ...props }/>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})