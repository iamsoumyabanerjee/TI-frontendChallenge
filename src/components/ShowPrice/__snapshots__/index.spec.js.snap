// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowPrice should render correctly when actual price and final price are not same 1`] = `
<div
  className="price-container price-sale"
>
  <span
    className="actual-price"
  >
    $
    100
  </span>
  <span
    className="sale-price"
  >
    $
    80
  </span>
</div>
`;

exports[`ShowPrice should render correctly when actual price and final price are same 1`] = `
<div
  className="price-container "
>
  <span
    className="actual-price"
  >
    $
    100
  </span>
</div>
`;
